var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -195,7 +182,7 @@ def determine_bootstrap_options(self, spec, withLibs, options):
         options.append("--with-libraries=%s" % ','.join(withLibs))
 
         if '+python' in spec:
-            options.append('--with-python=%s' % python_exe)
+            options.append('--with-python=%s' % spec['python'].executable)
```

Comment:
When installing with python as an external package, spack complains 'executable' is not defined in 'spec' of type 'Spec'.